src/Contract/Rector/RectorInterface.php

```diff
@@ -21,7 +20,16 @@ public function getNodeTypes(): array;
 
     /**
      * Process Node of matched type
-     * @return Node|Node[]|null|NodeTraverser::*
+     * @return Node|Node[]|null|int
+     *
+     * For int return, choose:
+     *
+     *   ✔️ To decorate current node and its children to not be traversed on current rule, return one of:
+     *          - NodeVisitor::DONT_TRAVERSE_CHILDREN
+     *          - NodeVisitor::DONT_TRAVERSE_CURRENT_AND_CHILDREN
+     *
+     *   ✔️ To remove node of Node\Stmt or Node\Param, return:
+     *          - NodeVisitor::REMOVE_NODE
      */
     public function refactor(Node $node);
```

Comment:
Is there any reason why this method does not have native return types?

```suggestion
     *          - NodeVisitor::REMOVE_NODE
    */
public function refactor(Node $node): array|Node|int|null;
```

Reply:
that is bc break, see https://3v4l.org/TPZ1l